### ISSUE TITLE
using adb shell enter key works just partially

### ISSUE THREAD

**Issue by helguson** [ubports/ubuntu-touch#811]
- Devices: FP2 (+ PC)
- Channel: 16.04 rc (+ 16.04 LTS on PC)
- Build: 2018-W31

### The problem
When being connected to the shell via `adb shell` then pressing `enter` key works only partially.
- pressing `enter` works for executing commands on the command line
- pressing `enter` does not work for using e.g., `nano` or `emacs`

### Steps to reproduce

1. run `adb shell` on PC
2. run `showkey -a`
3. press `enter`
4. press `CTRL-M`
5. press `ALT-M`
6. press `m`
7. press `CTRL-D` to stop `showkey`

### Expected behavior

Similar output like on PC or on the phone
- on the phone (using bluetooth keyboard)
  <img src="https://user-images.githubusercontent.com/3305305/43679344-f88d0d8a-9823-11e8-838d-db22e0498c83.png" width="300" >
- on pc
  <img src="https://user-images.githubusercontent.com/3305305/43679353-1e92e298-9824-11e8-8536-47e8e4c037cb.png" width="700" >

### Actual behavior

- over `adb shell` different output for `enter` and `CTRL-M`
  <img src="https://user-images.githubusercontent.com/3305305/43679578-56d81616-9827-11e8-8e39-3e1bbce751ab.png" width="700" >

### Additional information
This `CTRL-J` is also what e.g., `nano` and `emacs` receive and what they handle differently than `enter` / `CTRL-M`.

**Comment by Flohack74** [comment]
Hmm thats probably a problem with ADB itself, and not much we can do about. ADB is a debug bridge and not a fully flegded terminal protocol. A lot of stuff is not working via ADB, maybe also the key mapping as you describe. For that to work on we would need confirmation that it is working in Android :)

**Comment by dobey** [comment]
Indeed, ADB is not a full terminal emulator, and so many things do not work as you would expect them to in a normal terminal. If you want normal behavior, you should use SSH instead, or _phablet-shell_ script, which does SSH over ADB.

**Comment by helguson** [comment]
[They](https://sturmflut.github.io/ubuntu/touch/2015/05/08/hacking-ubuntu-touch-part-5-adb-shell-vs-phablet-shell/#adb-shell) describe this problem already in 2015:
> Try editing a file with nano, you can’t e.g. press the Enter key after typing in the filename the file should be saved to.

**Comment by helguson** [comment]
I guess it would be useful to present and compare these options for shell access in the [docs](http://docs.ubports.com).

**Comment by doniks** [comment]
@helguson adb and ssh are there http://docs.ubports.com/en/latest/userguide/advanceduse/index.html 
Not sure phablet-shell (still) exists for xenial (currently) 

merge requests to improve the docs are obviously welcome for what makes sense. Maybe some expectation management that adb is a *debug* tool and has it's limitations.
